libtsuba/include/katana/TxnContext.h

```diff
@@ -9,40 +9,48 @@ namespace katana {
 
 class KATANA_EXPORT TxnContext {
 public:
-  void InsertNodePropertyRead(std::string name) {
-    node_properties_read_.insert(name);
+  void InsertNodePropertyRead(std::string rdg_dir, std::string name) {
+    node_properties_read_.insert(ConcatRDGProperty(rdg_dir, name));
   }
 
   template <typename Container>
-  void InsertNodePropertyRead(const Container& names) {
-    node_properties_read_.insert(names.begin(), names.end());
+  void InsertNodePropertyRead(std::string rdg_dir, const Container& names) {
+    for (auto name : names) {
```

Comment:
Can all of these be `const auto&`

Reply:
I assume you will change these in your next PR?

Reply:
Hmmm sorry that I just saw this comment after merging. I'll add the changes in PR#831